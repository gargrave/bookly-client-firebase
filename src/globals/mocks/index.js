import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
