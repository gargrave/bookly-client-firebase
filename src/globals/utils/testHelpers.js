import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
