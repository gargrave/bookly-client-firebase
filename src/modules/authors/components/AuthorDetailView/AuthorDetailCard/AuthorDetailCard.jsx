// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Author } from '../../../flowtypes';

import Button from '../../../../common/components/Button/Button';
import ButtonRow from '../../../../common/components/ButtonRow/ButtonRow';
import Card from '../../../../common/components/Card/Card';

type Props = {
  author: Author,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
};

const authorDatesTextList = (author: Author) => {
  const {
    created,
    updated,
  } = author;

  return [
    { title: 'Added', value: format(created, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Updated', value: format(updated, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};

const AuthorDetailCard = ({
  author,
  onBackClick,
  onDeleteClick,
  onEditClick,
}: Props) => {
  return (
    <Card
      header={`${author.firstName} ${author.lastName}`}>

      <Card.Spacer />
      <Card.TextList textList={authorDatesTextList(author)} />
      <Card.Spacer size='large' />

      <ButtonRow>
        <Button
          onClick={onEditClick}
          position="left"
          text="Edit"
          type="info" />
        <Button
          onClick={onDeleteClick}
          text="Delete"
          type="danger" />
        <Button
          onClick={onBackClick}
          position="right"
          text="Back"
          type="light" />
      </ButtonRow>
    </Card>
  );
};

AuthorDetailCard.propTypes = {
  author: shape({
    id: string,
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    firstName: string,
    lastName: string,
  }),
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
};

export default AuthorDetailCard;
