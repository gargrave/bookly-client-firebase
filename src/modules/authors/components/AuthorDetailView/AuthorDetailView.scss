@import '../../../../styles/_views';

.authorDetailView {
  @include viewWrapper;
}
