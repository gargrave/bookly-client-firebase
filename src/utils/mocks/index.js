import authorMocks from './data/authors';
import userMocks from './data/users';

export {
  authorMocks,
  userMocks,
};
