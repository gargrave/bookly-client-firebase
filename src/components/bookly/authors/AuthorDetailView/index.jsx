// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Author } from '../../../../constants/flowtypes';

import { buildClasses } from '../../../../utils/cssHelpers';

import Alert from '../../../common/Alert';
import Button from '../../../common/Button';
import ButtonRow from '../../../common/ButtonRow';
import Card from '../../../common/Card';

type Props = {
  author: Author,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
  topLevelError?: string,
};

function AuthorDetailView({
  author,
  onBackClick,
  onDeleteClick,
  onEditClick,
  topLevelError,
}: Props) {
  return (
    <div className={buildClasses('author-detail-view')}>
      {topLevelError &&
        <Alert
          message={topLevelError}
          type="danger"
        />
      }
      <Card
        classes={['card--top-margin-med', 'detail-card', 'author-detail-card']}
        hoverable={false}
        title={`${author.firstName} ${author.lastName}`}
      >

        <hr/>
        <p className={buildClasses('card-text')}>
          <strong>Added:</strong> {format(author.created, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
        <p className={buildClasses('card-text')}>
          <strong>Updated:</strong> {format(author.updated, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>

        <hr/>

        <ButtonRow>
          <Button
            onClick={onEditClick}
            position="left"
            text="Edit"
            type="info"
          />
          <Button
            onClick={onDeleteClick}
            text="Delete"
            type="danger"
          />
          <Button
            onClick={onBackClick}
            position="right"
            text="Back"
            type="light"
          />
        </ButtonRow>
      </Card>
    </div>
  );
}

AuthorDetailView.propTypes = {
  author: shape({
    id: string,
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    firstName: string,
    lastName: string,
  }),
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
  topLevelError: string,
};

export default AuthorDetailView;
