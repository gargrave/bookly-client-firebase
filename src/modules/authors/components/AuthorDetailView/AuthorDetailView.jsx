// @flow
import React from 'react'
import { array, func, instanceOf, oneOfType, shape, string } from 'prop-types'
import styled from 'react-emotion'

import type { Author } from '../../flowtypes'
import type { Book } from '../../../books/flowtypes'

import { views } from '../../../../styles'

import Alert from '../../../common/components/Alert/Alert'
import AuthorBooksList from './AuthorBooksList/AuthorBooksList'
import AuthorDetailCard from './AuthorDetailCard/AuthorDetailCard'

type Props = {
  author: Author,
  booksForAuthor: Book[],
  onBackClick: Function,
  onBookClick: Function,
  onBookAddClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
  topLevelError?: string,
}

const Styled = styled('div')`
  ${views.viewWrapper};
`

const AuthorDetailView = ({
  author,
  booksForAuthor,
  onBackClick,
  onBookAddClick,
  onBookClick,
  onDeleteClick,
  onEditClick,
  topLevelError,
}: Props) => (
  <Styled>
    {topLevelError && <Alert message={topLevelError} type="danger" />}

    <AuthorDetailCard
      author={author}
      onBackClick={onBackClick}
      onDeleteClick={onDeleteClick}
      onEditClick={onEditClick}
    />

    <AuthorBooksList
      author={author}
      books={booksForAuthor}
      onBookAddClick={onBookAddClick}
      onBookClick={onBookClick}
    />
  </Styled>
)

AuthorDetailView.propTypes = {
  author: shape({
    id: string,
    created: oneOfType([instanceOf(Date), string]),
    updated: oneOfType([instanceOf(Date), string]),
    firstName: string,
    lastName: string,
  }),
  booksForAuthor: array.isRequired,
  onBackClick: func.isRequired,
  onBookClick: func.isRequired,
  onBookAddClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
  topLevelError: string,
}

export default AuthorDetailView
