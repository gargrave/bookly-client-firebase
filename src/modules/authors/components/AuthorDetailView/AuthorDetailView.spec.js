import { ComponentBuilder } from '../../../../utils/testHelpers';

import { authorMocks, bookMocks } from '../../../../globals/mocks/';

import AuthorDetailCard from './AuthorDetailCard/AuthorDetailCard';

import AuthorDetailView from './AuthorDetailView';

const defaultProps = {
  author: { ...authorMocks[0] },
  booksForAuthor: bookMocks,
  onBackClick: jest.fn(),
  onBookClick: jest.fn(),
  onBookAddClick: jest.fn(),
  onDeleteClick: jest.fn(),
  onEditClick: jest.fn(),
  topLevelError: '',
};

const builder = new ComponentBuilder(
  AuthorDetailView, defaultProps,
);

describe('AuthorDetailView', () => {
  let component;

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    component = builder.shallowGetComponent();
    expect(component.find('.bookly-author-detail-view')).toHaveLength(1);
    expect(component.find(AuthorDetailCard)).toHaveLength(1);
  });
});
