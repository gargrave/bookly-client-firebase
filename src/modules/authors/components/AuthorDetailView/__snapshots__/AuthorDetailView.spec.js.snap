// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDetailView matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetailView
    author={
      Object {
        "bookCount": 1,
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Aldous",
        "id": "oiw223h3qndlksadffl3ksdf",
        "lastName": "Huxley",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    booksForAuthor={
      Array [
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-14T01:36:26.220Z,
          "id": "sv3hmbefpc95MRXQsTHr",
          "sortBy": "",
          "title": "Brave New World",
          "updated": 2018-01-14T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-15T01:36:26.220Z,
          "id": "Hmh9eR3rs5bMvsTXfpcQ",
          "sortBy": "",
          "title": "Island",
          "updated": 2018-01-15T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "created": 2018-01-16T01:36:26.220Z,
          "id": "eRpcQ5H3rsTXfmh9bMvs",
          "sortBy": "",
          "title": "The Doors of Perception",
          "updated": 2018-01-16T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-06T22:02:11.504Z,
            "firstName": "Kurt",
            "id": "9328fhs938hfdsdblkhf98",
            "lastName": "Vonnegut",
            "updated": 2017-07-06T22:02:11.504Z,
          },
          "created": 2017-11-01T01:36:26.220Z,
          "id": "e2TedzRQJKJ1VL5m7QhG",
          "sortBy": "",
          "title": "Mother Night",
          "updated": 2017-11-01T01:36:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-06T22:02:11.504Z,
            "firstName": "Kurt",
            "id": "9328fhs938hfdsdblkhf98",
            "lastName": "Vonnegut",
            "updated": 2017-07-06T22:02:11.504Z,
          },
          "created": 2017-11-01T01:39:26.220Z,
          "id": "8ZYttrQY042IhlZ11BVJ",
          "sortBy": "",
          "title": "Breakfast of Champions",
          "updated": 2017-11-01T01:39:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 2,
            "created": 2017-07-06T22:02:17.051Z,
            "firstName": "Stephen",
            "id": "93ucbe8c8fn98w7b276v",
            "lastName": "King",
            "updated": 2017-08-17T21:27:35.885Z,
          },
          "created": 2017-10-01T01:39:26.220Z,
          "id": "2IYttrQYhlZ118Z04BVJ",
          "sortBy": "Dark Tower 1",
          "title": "The Gunslinger",
          "updated": 2017-10-01T01:39:26.220Z,
        },
        Object {
          "author": Object {
            "bookCount": 2,
            "created": 2017-07-06T22:02:17.051Z,
            "firstName": "Stephen",
            "id": "93ucbe8c8fn98w7b276v",
            "lastName": "King",
            "updated": 2017-08-17T21:27:35.885Z,
          },
          "created": 2017-10-01T01:39:26.220Z,
          "id": "2IYt04BVltrQYhZ118ZJ",
          "sortBy": "Dark Tower 2",
          "title": "The Drawing of the Three",
          "updated": 2017-10-01T01:39:26.220Z,
        },
      ]
    }
    onBackClick={[Function]}
    onBookAddClick={[Function]}
    onBookClick={[Function]}
    onDeleteClick={[Function]}
    onEditClick={[Function]}
    topLevelError=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "",
        <AuthorDetailCard
          author={
            Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          onBackClick={[Function]}
          onDeleteClick={[Function]}
          onEditClick={[Function]}
        />,
        <AuthorBooksList
          author={
            Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          books={
            Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "sortBy": "",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "sortBy": "",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "sortBy": "",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:36:26.220Z,
                "id": "e2TedzRQJKJ1VL5m7QhG",
                "sortBy": "",
                "title": "Mother Night",
                "updated": 2017-11-01T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:39:26.220Z,
                "id": "8ZYttrQY042IhlZ11BVJ",
                "sortBy": "",
                "title": "Breakfast of Champions",
                "updated": 2017-11-01T01:39:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 2,
                  "created": 2017-07-06T22:02:17.051Z,
                  "firstName": "Stephen",
                  "id": "93ucbe8c8fn98w7b276v",
                  "lastName": "King",
                  "updated": 2017-08-17T21:27:35.885Z,
                },
                "created": 2017-10-01T01:39:26.220Z,
                "id": "2IYttrQYhlZ118Z04BVJ",
                "sortBy": "Dark Tower 1",
                "title": "The Gunslinger",
                "updated": 2017-10-01T01:39:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 2,
                  "created": 2017-07-06T22:02:17.051Z,
                  "firstName": "Stephen",
                  "id": "93ucbe8c8fn98w7b276v",
                  "lastName": "King",
                  "updated": 2017-08-17T21:27:35.885Z,
                },
                "created": 2017-10-01T01:39:26.220Z,
                "id": "2IYt04BVltrQYhZ118ZJ",
                "sortBy": "Dark Tower 2",
                "title": "The Drawing of the Three",
                "updated": 2017-10-01T01:39:26.220Z,
              },
            ]
          }
          onBookAddClick={[Function]}
          onBookClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "onBackClick": [Function],
          "onDeleteClick": [Function],
          "onEditClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "books": Array [
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "sortBy": "",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-15T01:36:26.220Z,
              "id": "Hmh9eR3rs5bMvsTXfpcQ",
              "sortBy": "",
              "title": "Island",
              "updated": 2018-01-15T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-16T01:36:26.220Z,
              "id": "eRpcQ5H3rsTXfmh9bMvs",
              "sortBy": "",
              "title": "The Doors of Perception",
              "updated": 2018-01-16T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-06T22:02:11.504Z,
                "firstName": "Kurt",
                "id": "9328fhs938hfdsdblkhf98",
                "lastName": "Vonnegut",
                "updated": 2017-07-06T22:02:11.504Z,
              },
              "created": 2017-11-01T01:36:26.220Z,
              "id": "e2TedzRQJKJ1VL5m7QhG",
              "sortBy": "",
              "title": "Mother Night",
              "updated": 2017-11-01T01:36:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-06T22:02:11.504Z,
                "firstName": "Kurt",
                "id": "9328fhs938hfdsdblkhf98",
                "lastName": "Vonnegut",
                "updated": 2017-07-06T22:02:11.504Z,
              },
              "created": 2017-11-01T01:39:26.220Z,
              "id": "8ZYttrQY042IhlZ11BVJ",
              "sortBy": "",
              "title": "Breakfast of Champions",
              "updated": 2017-11-01T01:39:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 2,
                "created": 2017-07-06T22:02:17.051Z,
                "firstName": "Stephen",
                "id": "93ucbe8c8fn98w7b276v",
                "lastName": "King",
                "updated": 2017-08-17T21:27:35.885Z,
              },
              "created": 2017-10-01T01:39:26.220Z,
              "id": "2IYttrQYhlZ118Z04BVJ",
              "sortBy": "Dark Tower 1",
              "title": "The Gunslinger",
              "updated": 2017-10-01T01:39:26.220Z,
            },
            Object {
              "author": Object {
                "bookCount": 2,
                "created": 2017-07-06T22:02:17.051Z,
                "firstName": "Stephen",
                "id": "93ucbe8c8fn98w7b276v",
                "lastName": "King",
                "updated": 2017-08-17T21:27:35.885Z,
              },
              "created": 2017-10-01T01:39:26.220Z,
              "id": "2IYt04BVltrQYhZ118ZJ",
              "sortBy": "Dark Tower 2",
              "title": "The Drawing of the Three",
              "updated": 2017-10-01T01:39:26.220Z,
            },
          ],
          "onBookAddClick": [Function],
          "onBookClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "",
          <AuthorDetailCard
            author={
              Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            onBackClick={[Function]}
            onDeleteClick={[Function]}
            onEditClick={[Function]}
          />,
          <AuthorBooksList
            author={
              Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            books={
              Array [
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-14T01:36:26.220Z,
                  "id": "sv3hmbefpc95MRXQsTHr",
                  "sortBy": "",
                  "title": "Brave New World",
                  "updated": 2018-01-14T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-15T01:36:26.220Z,
                  "id": "Hmh9eR3rs5bMvsTXfpcQ",
                  "sortBy": "",
                  "title": "Island",
                  "updated": 2018-01-15T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-01T01:16:43.881Z,
                    "firstName": "Aldous",
                    "id": "oiw223h3qndlksadffl3ksdf",
                    "lastName": "Huxley",
                    "updated": 2017-07-01T01:16:43.881Z,
                  },
                  "created": 2018-01-16T01:36:26.220Z,
                  "id": "eRpcQ5H3rsTXfmh9bMvs",
                  "sortBy": "",
                  "title": "The Doors of Perception",
                  "updated": 2018-01-16T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-06T22:02:11.504Z,
                    "firstName": "Kurt",
                    "id": "9328fhs938hfdsdblkhf98",
                    "lastName": "Vonnegut",
                    "updated": 2017-07-06T22:02:11.504Z,
                  },
                  "created": 2017-11-01T01:36:26.220Z,
                  "id": "e2TedzRQJKJ1VL5m7QhG",
                  "sortBy": "",
                  "title": "Mother Night",
                  "updated": 2017-11-01T01:36:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 1,
                    "created": 2017-07-06T22:02:11.504Z,
                    "firstName": "Kurt",
                    "id": "9328fhs938hfdsdblkhf98",
                    "lastName": "Vonnegut",
                    "updated": 2017-07-06T22:02:11.504Z,
                  },
                  "created": 2017-11-01T01:39:26.220Z,
                  "id": "8ZYttrQY042IhlZ11BVJ",
                  "sortBy": "",
                  "title": "Breakfast of Champions",
                  "updated": 2017-11-01T01:39:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 2,
                    "created": 2017-07-06T22:02:17.051Z,
                    "firstName": "Stephen",
                    "id": "93ucbe8c8fn98w7b276v",
                    "lastName": "King",
                    "updated": 2017-08-17T21:27:35.885Z,
                  },
                  "created": 2017-10-01T01:39:26.220Z,
                  "id": "2IYttrQYhlZ118Z04BVJ",
                  "sortBy": "Dark Tower 1",
                  "title": "The Gunslinger",
                  "updated": 2017-10-01T01:39:26.220Z,
                },
                Object {
                  "author": Object {
                    "bookCount": 2,
                    "created": 2017-07-06T22:02:17.051Z,
                    "firstName": "Stephen",
                    "id": "93ucbe8c8fn98w7b276v",
                    "lastName": "King",
                    "updated": 2017-08-17T21:27:35.885Z,
                  },
                  "created": 2017-10-01T01:39:26.220Z,
                  "id": "2IYt04BVltrQYhZ118ZJ",
                  "sortBy": "Dark Tower 2",
                  "title": "The Drawing of the Three",
                  "updated": 2017-10-01T01:39:26.220Z,
                },
              ]
            }
            onBookAddClick={[Function]}
            onBookClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "onBackClick": [Function],
            "onDeleteClick": [Function],
            "onEditClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "books": Array [
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "sortBy": "",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "sortBy": "",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "sortBy": "",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:36:26.220Z,
                "id": "e2TedzRQJKJ1VL5m7QhG",
                "sortBy": "",
                "title": "Mother Night",
                "updated": 2017-11-01T01:36:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-06T22:02:11.504Z,
                  "firstName": "Kurt",
                  "id": "9328fhs938hfdsdblkhf98",
                  "lastName": "Vonnegut",
                  "updated": 2017-07-06T22:02:11.504Z,
                },
                "created": 2017-11-01T01:39:26.220Z,
                "id": "8ZYttrQY042IhlZ11BVJ",
                "sortBy": "",
                "title": "Breakfast of Champions",
                "updated": 2017-11-01T01:39:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 2,
                  "created": 2017-07-06T22:02:17.051Z,
                  "firstName": "Stephen",
                  "id": "93ucbe8c8fn98w7b276v",
                  "lastName": "King",
                  "updated": 2017-08-17T21:27:35.885Z,
                },
                "created": 2017-10-01T01:39:26.220Z,
                "id": "2IYttrQYhlZ118Z04BVJ",
                "sortBy": "Dark Tower 1",
                "title": "The Gunslinger",
                "updated": 2017-10-01T01:39:26.220Z,
              },
              Object {
                "author": Object {
                  "bookCount": 2,
                  "created": 2017-07-06T22:02:17.051Z,
                  "firstName": "Stephen",
                  "id": "93ucbe8c8fn98w7b276v",
                  "lastName": "King",
                  "updated": 2017-08-17T21:27:35.885Z,
                },
                "created": 2017-10-01T01:39:26.220Z,
                "id": "2IYt04BVltrQYhZ118ZJ",
                "sortBy": "Dark Tower 2",
                "title": "The Drawing of the Three",
                "updated": 2017-10-01T01:39:26.220Z,
              },
            ],
            "onBookAddClick": [Function],
            "onBookClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
